File: src/main/scala/com/sh/utils/HiveUtils.scala
package com.sh.utils

import java.sql.DriverManager

import org.apache.spark.{SparkConf, SparkContext}
import org.apache.spark.sql.hive.HiveContext

object HiveUtils {
  def main(args: Array[String]): Unit = {
//    val tablename = "hive_bdtable";
//
//    val HIVE_SERVER = "jdbc:hive2://localhost:10000/default"
//    Class forName("org.apche.hive.jdbc.hiveDriver")
//    val conn=DriverManager.getConnection(HIVE_SERVER,"hadoop", "hive")
//    val stmt =conn.createStatement()
//    val addSQL="select * from " + tablename
//    val res = stmt.execute(addSQL)
//    conn.close()



    println("*****************")
    val conf = new SparkConf().setAppName("HiveUtils").setMaster("local[2]")
    val sc = new SparkContext(conf)
    val sqlContext = new HiveContext(sc)

    val hivecontext= new HiveContext(sc) //再把配置好的上下文对象作为参数传给hive 目的是针对整个APP执行sql
    val dataframeHive = hivecontext.table("hive_bdtable")
    dataframeHive.show()

//    sqlContext.table("hive.hive_bdtable") // 库名.表名 的格式
//      .registerTempTable("hive_bdtable")  // 注册成临时表
//    sqlContext.sql(
//      """
//        | select *
//        |   from hive_bdtable
//        |  limit 10
//      """.stripMargin).show()

    sc.stop()
  }

//  def dropPartitions(tablename:String,hivecotext:HiveContext):Unit = {
//
//    sqlContext.sql()
//
//  }
}

File: src/main/java/com/sh/java/utils/HiveUtils.java
package com.sh.java.utils;

import java.sql.*;

public class HiveUtils {

    // 备注，不是private static String driverName = "org.apache.hadoop.hive.jdbc.HiveDriver";
    private static String driverName = "org.apache.hive.jdbc.HiveDriver";

    public static void main(String[] args) throws SQLException, ClassNotFoundException {
        Class.forName(driverName);
        // 备注：不是jdbc:hive://localhost:10000/default", "", ""
        Connection con = DriverManager.getConnection("jdbc:hive2://localhost:10000/default", "hadoop", "hive");// 此处的hadoop是服务器对应的user的name名称；hive是表名
        Statement stmt = con.createStatement();

        String tableName = "hive_bdtable";
        // select * query
        String sql = "select * from " + tableName;
        System.out.println("Running: " + sql);
        ResultSet res = stmt.executeQuery(sql);
        if (res.next()) {
            System.out.println(res.getString(1));
        }
        // select * query
        sql = "select * from " + tableName;
        System.out.println("Running: " + sql);
        res = stmt.executeQuery(sql);
        while (res.next()) {
            System.out.println(String.valueOf(res.getString(1)) + " " + res.getDouble(2) + " " + res.getDouble(3));
        }


        System.out.println("Database userdb created successfully.");

        con.close();
    }
}
